### PR TITLE
Add deregister broker command to svcat

### PR DESCRIPTION
**Pull request by jberkhahn**
#1807

Simple first take on it, will add fancier flags later.

@carolynvs please take a look

### CODE REVIEW

**Comment by jeremyrickard** [comment]
LGTM

File: cmd/svcat/broker/deregister_cmd.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	"github.com/spf13/cobra"
+)
+
+// DeregisterCmd contains the info needed to delete a broker
+type DeregisterCmd struct {
+	BrokerName string
+	Context    *command.Context
+}
+
+// NewDeregisterCmd builds a "svcat deegister" command
```
Comment (carolynvs) [inline cmd/svcat/broker/deregister_cmd.go]: typo: `svcat deregister`

File: cmd/svcat/broker/deregister_cmd.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+// Deregister calls out to the pkg lib to delete the broker and display the output
+func (c *DeregisterCmd) Deregister() error {
+	err := c.Context.App.Deregister(c.BrokerName)
+	if err != nil {
+		return err
+	}
+
+	outputBytes := []byte(fmt.Sprintf("Successfully removed ClusterServiceBroker \"%v\"", c.BrokerName))
```
Comment (carolynvs) [inline cmd/svcat/broker/deregister_cmd.go]: You can shorten this to `fmt.Fprintf(c.Output, "Successfully removed ClusterServiceBroker %q", c.BrokerName)`

Comment (carolynvs) [reply cmd/svcat/broker/deregister_cmd.go]: nit: I've been avoiding using our type names in the svcat output and pick a user friendly name whenever possible. So I'd prefer it if we didn't go with a type name here. Maybe just go with `Successfully removed broker "foo"` instead, so that when this works for namespaced brokers we can use `Successfully removed broker "foo" from namespace "bar"`.

File: cmd/svcat/broker/deregister_cmd_test.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+var _ = Describe("Deregister Command", func() {
+	Describe("NewDeregisterCmd", func() {
+		It("Builds and returns a cobra command", func() {
+			cxt := &command.Context{}
+			cmd := NewDeregisterCmd(cxt)
+			Expect(*cmd).NotTo(BeNil())
+			Expect(cmd.Use).To(Equal("deregister NAME"))
+			Expect(cmd.Short).To(Equal("Deregisters an existing broker with service catalog"))
+			Expect(cmd.Example).To(Equal("  svcat deregister mysqlbroker"))
```
Comment (carolynvs) [inline cmd/svcat/broker/deregister_cmd_test.go]: Mind switching this to a contains we are aren't checking the indent level too on this?

File: cmd/svcat/broker/deregister_cmd_test.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		})
+	})
+	Describe("Validate", func() {
+		It("errors if a broker name is not provided", func() {
+			cmd := DeregisterCmd{
+				BrokerName: "",
+				Context:    nil,
+			}
+			err := cmd.Validate([]string{})
+			Expect(err).To(HaveOccurred())
```
Comment (carolynvs) [inline cmd/svcat/broker/deregister_cmd_test.go]: Is it possible to assert that the error message contains an expected substring?

File: pkg/svcat/service-catalog/broker_test.go [hunk 5]
```diff
@@ -48,6 +48,33 @@ var _ = Describe("Broker", func() {
 		}
 	})
 
+	Describe("Deregister", func() {
+		It("deletesa broker by calling the v1beta1 Delete method with the passed in arguement", func() {
```
Comment (carolynvs) [inline pkg/svcat/service-catalog/broker_test.go]: typo: "deletes a"

**Comment by jberkhahn** [comment]
Fixed Carolyn's suggestions.

**Comment by carolynvs** [comment]
Carolyn: 1  GitHub: 0
